### PR TITLE
Global Index Calculation for Multiple nodal d.o.f.

### PR DESCRIPTION
**Pull request by chleh**
VectorMatrixAssembler now has a pointer to the variables the global problem is solved for.
The indices of a variable in the global matrix are now computed based on the setting BY_COMPONENT/BY_LOCATION.
However, I changed VectorMatrixAssembler such that in the local matrices are always ordered by component.

### CODE REVIEW

**Comment by bilke** [comment]
Failed Mac build can be ignored!

**Comment by endJunction** [comment]
Jenkins, test this please.

**Comment by endJunction** [comment]
@chleh Please have a look on the [multiComponentDirichletBCs](https://github.com/endJunction/ogs/tree/multiComponentDirichletBCs) branch for the first split-off of this PR.

File: AssemblerLib/MeshComponentMap.cpp [hunk 1]
```diff
@@ -189,4 +191,34 @@ MeshComponentMap::getGlobalIndices<ComponentOrder::BY_COMPONENT>(
     return global_indices;
 }
 
+std::vector<std::size_t>
+MeshComponentMap::getIndicesForComponent(
```
Comment (wenqing) [inline AssemblerLib/MeshComponentMap.cpp]: I added two functions that have the same functionality in PR #580.

File: ProcessLib/GroundwaterFlowFEM.h [hunk 2]
```diff
@@ -34,7 +34,8 @@ class LocalAssemblerDataInterface
             Parameter<double, MeshLib::Element const&> const& hydraulic_conductivity,
             unsigned const integration_order) = 0;
 
-    virtual void assemble() = 0;
+    virtual void assemble(std::vector<double> const& local_x,
```
Comment (wenqing) [inline ProcessLib/GroundwaterFlowFEM.h]: indentation.

**Pull request closed without merging**